pkg/storage/driver/sql.go

```diff
@@ -0,0 +1,330 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package driver
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	migrate "github.com/rubenv/sql-migrate"
+
+	// Import pq for potgres dialect
+	_ "github.com/lib/pq"
+
+	rspb "k8s.io/helm/pkg/proto/hapi/release"
+	storageerrors "k8s.io/helm/pkg/storage/errors"
+)
+
+var _ Driver = (*SQL)(nil)
+
+var labelMap = map[string]string{
+	"MODIFIED_AT": "modified_at",
+	"CREATED_AT":  "created_at",
+	"VERSION":     "version",
+	"STATUS":      "status",
+	"OWNER":       "owner",
+	"NAME":        "name",
+}
+
+var supportedSQLDialects = map[string]struct{}{
+	"postgres": struct{}{},
+}
+
+// SQLDriverName is the string name of this driver.
+const SQLDriverName = "SQL"
+
+// SQL is the sql storage driver implementation.
+type SQL struct {
+	db  *sqlx.DB
+	Log func(string, ...interface{})
+}
+
+// Name returns the name of the driver.
+func (s *SQL) Name() string {
+	return SQLDriverName
+}
+
+func (s *SQL) ensureDBSetup() error {
+	// Populate the database with the relations we need if they don't exist yet
+	migrations := &migrate.MemoryMigrationSource{
+		Migrations: []*migrate.Migration{
+			{
+				Id: "init",
+				Up: []string{
+					`
+            CREATE TABLE releases (
```

Comment:
Nit: The whitespace here seems to be a mix of spaces and tabs for line prefixing.